src/ray/core_worker/core_worker.cc

```diff
@@ -1650,39 +1646,45 @@ std::shared_ptr<rpc::RuntimeEnvInfo> CoreWorker::OverrideTaskOrActorRuntimeEnvIn
 
   if (options_.worker_type == WorkerType::DRIVER) {
     if (IsRuntimeEnvEmpty(runtime_env_info->serialized_runtime_env())) {
-      return job_runtime_env_info_;
+      return std::make_shared<rpc::RuntimeEnvInfo>(
+          worker_context_.GetCurrentJobConfig().runtime_env_info());
```

Comment:
should we return the `std::shared_ptr<const rpc::RuntimeEnvInfo>`? Otherwise doesn's it mean the caller of this API can modify the runtime_env_info from the worker context?

Reply:
we can do it in a separate PR.